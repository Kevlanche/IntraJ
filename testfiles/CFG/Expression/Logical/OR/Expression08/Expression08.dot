digraph G {
graph [splines=ortho, nodesep="1", ranksep="1"]
node [shape=rect, fontname=Arial];
{ rank = same; "Program[0]"}
{ rank = same; "List[1]"}
{ rank = same; "CompilationUnit[2]"}
{ rank = same; "List[3]"; "List[4]"}
{ rank = same; "ClassDecl[5]"}
{ rank = same; "Modifiers[6]"; "Opt[9]"; "List[10]"; "List[11]"}
{ rank = same; "List[7]"; "MethodDecl[12]"}
{ rank = same; "Modifier[8]"; "Entry[13]"; "Modifiers[14]"; "PrimitiveTypeAccess[16]"; "List[17]"; "List[26]"; "Opt[27]"; "Exit[28]"}
{ rank = same; "List[15]"; "ParameterDeclaration[18]"; "ParameterDeclaration[22]"; "Block[28]"}
{ rank = same; "Modifiers[19]"; "PrimitiveTypeAccess[21]"; "Modifiers[23]"; "PrimitiveTypeAccess[25]"; "List[29]"}
{ rank = same; "List[20]"; "List[24]"; "ReturnStmt[30]"}
{ rank = same; "Opt[31]"; "List[39]"; "UncheckedExceptions[31]"}
{ rank = same; "OrLogicalExpr[32]"}
{ rank = same; "GTExpr[33]"; "LTExpr[36]"}
{ rank = same; "VarAccess[34]"; "IntegerLiteral[35]"; "VarAccess[37]"; "IntegerLiteral[38]"}
"Program[0]" [label="Program\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[1]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"CompilationUnit[2]" [label="CompilationUnit\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[3]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[4]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ClassDecl[5]" [label="ClassDecl\n < Expression08 >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[6]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[7]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifier[8]" [label="Modifier\n < public >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[9]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[10]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[11]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"MethodDecl[12]" [label="MethodDecl\n < m >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Entry[13]" [label="Entry\n ", fillcolor=white   style=filled  ]
"Modifiers[14]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[15]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[16]" [label="PrimitiveTypeAccess\n < boolean >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[17]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[18]" [label="ParameterDeclaration\n < a >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[19]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[20]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[21]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ParameterDeclaration[22]" [label="ParameterDeclaration\n < b >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Modifiers[23]" [label="Modifiers\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[24]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"PrimitiveTypeAccess[25]" [label="PrimitiveTypeAccess\n < int >", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[26]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Opt[27]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Block[28]" [label="Block\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"List[29]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"ReturnStmt[30]" [label="ReturnStmt\n ", fillcolor=white   style=filled  ]
"Opt[31]" [label="Opt\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"OrLogicalExpr[32]" [label="OrLogicalExpr\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"GTExpr[33]" [label="GTExpr\n ", fillcolor=white   style=filled  ]
"VarAccess[34]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"IntegerLiteral[35]" [label="IntegerLiteral\n < 10 >", fillcolor=white   style=filled  ]
"LTExpr[36]" [label="LTExpr\n ", fillcolor=white   style=filled  ]
"VarAccess[37]" [label="VarAccess\n < a >", fillcolor=white   style=filled  ]
"IntegerLiteral[38]" [label="IntegerLiteral\n < 10 >", fillcolor=white   style=filled  ]
"List[39]" [label="List\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"UncheckedExceptions[31]" [label="UncheckedExceptions\n ", style= dotted   fillcolor="#eeeeee" fontcolor="#aaaaaa"  ]
"Exit[28]" [label="Exit\n ", fillcolor=white   style=filled  ]
"CompilationUnit[2]" -> "List[3]" [style=dashed, color=gray]
"List[7]" -> "Modifier[8]" [style=dashed, color=gray]
"Modifiers[6]" -> "List[7]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Modifiers[6]" [style=dashed, color=gray]
"ClassDecl[5]" -> "Opt[9]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[10]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Entry[13]" [style=dashed, color=gray]
"Modifiers[14]" -> "List[15]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Modifiers[14]" [style=dashed, color=gray]
"MethodDecl[12]" -> "PrimitiveTypeAccess[16]" [style=dashed, color=gray]
"Modifiers[19]" -> "List[20]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "Modifiers[19]" [style=dashed, color=gray]
"ParameterDeclaration[18]" -> "PrimitiveTypeAccess[21]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[18]" [style=dashed, color=gray]
"Modifiers[23]" -> "List[24]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "Modifiers[23]" [style=dashed, color=gray]
"ParameterDeclaration[22]" -> "PrimitiveTypeAccess[25]" [style=dashed, color=gray]
"List[17]" -> "ParameterDeclaration[22]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[17]" [style=dashed, color=gray]
"MethodDecl[12]" -> "List[26]" [style=dashed, color=gray]
"GTExpr[33]" -> "VarAccess[34]" [style=dashed, color=gray]
"GTExpr[33]" -> "IntegerLiteral[35]" [style=dashed, color=gray]
"OrLogicalExpr[32]" -> "GTExpr[33]" [style=dashed, color=gray]
"LTExpr[36]" -> "VarAccess[37]" [style=dashed, color=gray]
"LTExpr[36]" -> "IntegerLiteral[38]" [style=dashed, color=gray]
"OrLogicalExpr[32]" -> "LTExpr[36]" [style=dashed, color=gray]
"Opt[31]" -> "OrLogicalExpr[32]" [style=dashed, color=gray]
"ReturnStmt[30]" -> "Opt[31]" [style=dashed, color=gray]
"ReturnStmt[30]" -> "List[39]" [style=dashed, color=gray]
"ReturnStmt[30]" -> "UncheckedExceptions[31]" [style=dashed, color=gray]
"List[29]" -> "ReturnStmt[30]" [style=dashed, color=gray]
"Block[28]" -> "List[29]" [style=dashed, color=gray]
"Opt[27]" -> "Block[28]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Opt[27]" [style=dashed, color=gray]
"MethodDecl[12]" -> "Exit[28]" [style=dashed, color=gray]
"List[11]" -> "MethodDecl[12]" [style=dashed, color=gray]
"ClassDecl[5]" -> "List[11]" [style=dashed, color=gray]
"List[4]" -> "ClassDecl[5]" [style=dashed, color=gray]
"CompilationUnit[2]" -> "List[4]" [style=dashed, color=gray]
"List[1]" -> "CompilationUnit[2]" [style=dashed, color=gray]
"Program[0]" -> "List[1]" [style=dashed, color=gray]
"Exit[28]" -> "ReturnStmt[30]" [color=red, constraint=false]
"ReturnStmt[30]" -> "Exit[28]" [color=blue, constraint=false, xlabel=" "] 
"ReturnStmt[30]" -> "GTExpr[33]" [color=red, constraint=false]
"ReturnStmt[30]" -> "LTExpr[36]" [color=red, constraint=false]
"LTExpr[36]" -> "ReturnStmt[30]" [color=blue, constraint=false, xlabel=" "] 
"LTExpr[36]" -> "IntegerLiteral[38]" [color=red, constraint=false]
"IntegerLiteral[38]" -> "LTExpr[36]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[38]" -> "VarAccess[37]" [color=red, constraint=false]
"VarAccess[37]" -> "IntegerLiteral[38]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[37]" -> "GTExpr[33]" [color=red, constraint=false]
"GTExpr[33]" -> "ReturnStmt[30]" [color=blue, constraint=false, xlabel="T "] 
"GTExpr[33]" -> "VarAccess[37]" [color=blue, constraint=false, xlabel="F "] 
"GTExpr[33]" -> "IntegerLiteral[35]" [color=red, constraint=false]
"IntegerLiteral[35]" -> "GTExpr[33]" [color=blue, constraint=false, xlabel=" "] 
"IntegerLiteral[35]" -> "VarAccess[34]" [color=red, constraint=false]
"VarAccess[34]" -> "IntegerLiteral[35]" [color=blue, constraint=false, xlabel=" "] 
"VarAccess[34]" -> "Entry[13]" [color=red, constraint=false]
"Entry[13]" -> "VarAccess[34]" [color=blue, constraint=false, xlabel=" "] 
"List[3]" -> "List[4]" [style=invis]
"Modifiers[6]" -> "Opt[9]" -> "List[10]" -> "List[11]" [style=invis]
"Entry[13]" -> "Modifiers[14]" -> "PrimitiveTypeAccess[16]" -> "List[17]" -> "List[26]" -> "Opt[27]" -> "Exit[28]" [style=invis]
"ParameterDeclaration[18]" -> "ParameterDeclaration[22]" [style=invis]
"Modifiers[19]" -> "PrimitiveTypeAccess[21]" [style=invis]
"Modifiers[23]" -> "PrimitiveTypeAccess[25]" [style=invis]
"GTExpr[33]" -> "LTExpr[36]" [style=invis]
"VarAccess[34]" -> "IntegerLiteral[35]" [style=invis]
"VarAccess[37]" -> "IntegerLiteral[38]" [style=invis]

subgraph cluster_legend {
{node [style=filled, fillcolor=1, colorscheme="pastel13"]
Nodes [label="Total Nr. Nodes
42", fillcolor=1]
Edges [label="Total Nr. Edges
59", fillcolor=1]
NodesCFG [label="CFG Nr. Nodes
9", fillcolor=3]
EdgesCFG [label="CFG Nr. Edges
9", fillcolor=3]
Nodes-> "Program[0]" -> NodesCFG  [style="invis"]
Edges -> "Program[0]" ->EdgesCFG [style="invis"]}
label = "Statistics of the framework: NEXTFramework"
style="solid"
ranksep=0.05
nodesep=0.01
labelloc = b
len=2
}}