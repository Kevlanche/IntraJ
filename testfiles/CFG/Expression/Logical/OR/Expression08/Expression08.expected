NodeName: Entry[13] Successors: [VarAccess[34]]
NodeName: VarAccess[34] Successors: [IntegerLiteral[35]]
NodeName: IntegerLiteral[35] Successors: [GTExpr[33]]
NodeName: GTExpr[33] Successors: [ReturnStmt[30], VarAccess[37]]
NodeName: ReturnStmt[30] Successors: [Exit[28]]
NodeName: Exit[28] Successors: []
NodeName: VarAccess[37] Successors: [IntegerLiteral[38]]
NodeName: IntegerLiteral[38] Successors: [LTExpr[36]]
NodeName: LTExpr[36] Successors: [ReturnStmt[30]]

NodeName: Exit[28] Predecessor: [ReturnStmt[30]]
NodeName: ReturnStmt[30] Predecessor: [GTExpr[33], LTExpr[36]]
NodeName: GTExpr[33] Predecessor: [IntegerLiteral[35]]
NodeName: IntegerLiteral[35] Predecessor: [VarAccess[34]]
NodeName: VarAccess[34] Predecessor: [Entry[13]]
NodeName: Entry[13] Predecessor: []
NodeName: LTExpr[36] Predecessor: [IntegerLiteral[38]]
NodeName: IntegerLiteral[38] Predecessor: [VarAccess[37]]
NodeName: VarAccess[37] Predecessor: [GTExpr[33]]
